object_storage_api/repositories/image.py

```diff
@@ -130,23 +131,27 @@ def update(self, image_id: str, image: ImageIn, update_primary: bool, session: C
             exc.response_detail = "Image not found"
             raise exc
 
-        stored_image = self.get(str(image_id), session=session)
-        if image.file_name != stored_image.file_name and self._is_duplicate(
-            image.entity_id, image.code, image_id, session=session
-        ):
-            raise DuplicateRecordError("Duplicate image found within the parent entity", entity_type="image")
-
-        logger.info("Updating image metadata with ID: %s", image_id)
-        if update_primary:
-            bulkwrite_update = [
-                UpdateMany(filter={"primary": True, "entity_id": image.entity_id}, update={"$set": {"primary": False}}),
-                UpdateOne(filter={"_id": image_id}, update={"$set": image.model_dump(by_alias=True)}),
-            ]
-            self._images_collection.bulk_write(bulkwrite_update, session=session)
-        else:
-            self._images_collection.update_one(
-                {"_id": image_id}, {"$set": image.model_dump(by_alias=True)}, session=session
-            )
+        try:
+            if update_primary:
+                bulkwrite_update = [
+                    UpdateOne(
+                        filter={"_id": image_id}, update={"$set": image.model_dump(by_alias=True, exclude="primary")}
```

Comment:
`Expected type 'set[int] | set[str] | Mapping[int, Literal[True] | Any] | Mapping[str, Literal[True] | Any] | None', got 'Literal["primary"]' instead`
```suggestion
                    UpdateOne(
                        filter={"_id": image_id}, update={"$set": image.model_dump(by_alias=True, exclude={"primary"})}
```